tests/cxx/unit_tests/xyz_to_mol.cpp

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2024 NWChemEx-Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "../test_nux.hpp"
+#include <nux/nux.hpp>
+
+TEST_CASE("XYZToMolecule") {
+    pluginplay::ModuleManager mm;
+    nux::load_modules(mm);
+
+    auto make_atoms = [=](auto&& Z) {
+      double h_mass = 1837.4260218693814;
+      return simde::type::atom{"H", 1, h_mass, 0.0, 0.0, 0.0};
+    };
+
+    auto atom_mod = pluginplay::make_lambda<simde::AtomFromZ>(make_atoms);
+
+    auto z_mod = pluginplay::make_lambda<simde::ZFromSymbol>([=](auto&& symbol) {
+        return simde::type::atomic_number{1};
+    });
+
+    auto& xyz_mod = mm.at("XYZ To Molecule");
+    xyz_mod.change_submod("Z from symbol", z_mod);
+    xyz_mod.change_submod("Atom from z", atom_mod);
+
+  SECTION("No XYZ file found") {
+    std::string file = "file.xyz";
+    REQUIRE_THROWS_AS(xyz_mod.run_as<simde::MoleculeFromString>(file), std::runtime_error);
+  }
```

Comment:
The formatting on this file is off. You'll need to copy the `.clang-format` file from one of the other repos in the stack and add it.